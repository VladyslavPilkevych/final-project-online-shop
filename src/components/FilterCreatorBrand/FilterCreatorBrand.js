import React, { memo, useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import styles from './FilterCreatorBrand.module.scss';
import {
  filterBrand,
} from '../../store/actionCreators/filterAC';

function FilterCreatorBrand(props) {
  const { brandsFiltered } = props;
  const dispatch = useDispatch();
  const filterByBrand = useSelector((state) => state.filter.filterByBrand);
  return (
    <ul className={styles.brandsContainer}>
      {brandsFiltered
                && brandsFiltered.map((brand) => (
                  <li key={brand} className={styles.brandsLi}>
                    <input
                      type="checkbox"
                      className={styles.brandsCheckbox}
                      id={`brand_${brand}`}
                      name={`brand_${brand}`}
                      checked={filterByBrand.includes(brand)}
                      onChange={() => {
                        const selectedBrands = [...filterByBrand];
                        if (!selectedBrands.includes(brand)) {
                          selectedBrands.push(brand);
                        } else {
                          const index = selectedBrands.findIndex((item) => item === brand);
                          selectedBrands.splice(index, 1);
                        }
                        dispatch(filterBrand(selectedBrands));
                      }}
                    />
                    <label htmlFor={`brand_${brand}`}>{brand}</label>
                  </li>
                ))}
    </ul>
  );
}

FilterCreatorBrand.propTypes = {
  brandsFiltered: PropTypes.array,
};

FilterCreatorBrand.defaultProps = {
  brandsFiltered: [],
};

export default memo(FilterCreatorBrand);
