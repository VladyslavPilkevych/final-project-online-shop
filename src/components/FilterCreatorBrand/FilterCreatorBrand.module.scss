.brandsCheckbox {
    margin: 4px 10px 10px 0;
}