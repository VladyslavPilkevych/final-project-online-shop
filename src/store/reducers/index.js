import { combineReducers } from 'redux';
import preloaderReducer from './preloaderReducer';
import productsReducer from './productsReducer';
import menuReducer from './menuReducer';
import userReducer from './userReducer';
import searchReducer from './searchReducer';
import cartReducer from './cartReducer';
import miniMenuReducer from './miniMenuReducer';
import filtersCategoriesReducer from './filtersCategoriesReducer';

const reducer = combineReducers({
  loader: preloaderReducer,
  products: productsReducer,
  menu: menuReducer,
  miniMenu: miniMenuReducer,
  filtersCategories: filtersCategoriesReducer,
  user: userReducer,
  search: searchReducer,
  cart: cartReducer,
});

export default reducer;
