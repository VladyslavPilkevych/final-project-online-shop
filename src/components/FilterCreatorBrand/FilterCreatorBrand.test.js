import React, { Fragment } from 'react';
// import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
// import { fireEvent, getByText, queryByTestId, queryByText, render, screen } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';
import { Provider, useDispatch } from 'react-redux';
import FilterCreatorBrand from './FilterCreatorBrand';
import store from '../../store';
// import Router from 'react-router-dom';
// import { Routes, Route, Navigate } from 'react-router-dom';

const brandsFiltered = ['apple', 'android', 'linux', 'windows'];

function Component() {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <FilterCreatorBrand brandsFiltered={brandsFiltered} />
      </Provider>
    </BrowserRouter>
  );
}

describe('Snapshot FilterCreatorBrand should render', () => {
  test('Snapshot FilterCreatorBrand is rendered', () => {
    const { asFragment } = render(<Component />);
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot(asFragment());
  });
});

describe('brand redux should work', () => {
  test('should add to store', () => {
    const result = render(<Component />);
    const brand = result.getByText('windows');
    brand.click();
    const checkbox = result.container.querySelector('#brand_windows');
    expect(checkbox).toBeChecked();
  });
  // test('should delete from store', () => {
  //     const result = render(<Component />);
  //     const brand = result.getByText('windows');
  //     brand.click();
  //     brand.click();
  //     const checkbox = result.container.querySelector('#brand_windows');
  //     expect(checkbox).not.toBeChecked();
  // });
});
