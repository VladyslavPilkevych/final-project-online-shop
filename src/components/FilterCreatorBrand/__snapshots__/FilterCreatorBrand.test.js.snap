// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot FilterCreatorBrand should render Snapshot FilterCreatorBrand is rendered 1`] = `Object {}`;
